### DISCUSSION TITLE
Creating a bucket with access set to public with Amazon S3 Java SDK

### DISCUSSION THREAD

**Discussion by minthaka** [minio/minio#15884] [Q&A] [answered]
Hello!
I am trying to create a bucket via Amazon API, with the following code:

/* Create an Object of CreateBucketRequest */
                CreateBucketRequest request = new CreateBucketRequest("mybucket");
                 
                /* Set Canned ACL as PublicRead */
                request.setCannedAcl(CannedAccessControlList.PublicRead);
                 
                /* Send Create Bucket Request */
                Bucket result = s3.createBucket(request);

It does create a bucket called mybucket on Minio server, but the access policy is still set to private, so I am not able to download the files from this bucket with an anonym user from a web browser.

Where do I am wrong? Is it possible to accomplish this only using API?

**Comment by harshavardhana** [comment, marked as answer]
There are not ACLs on MinIO use Bucket Policy to give anonymous access to your bucket objects.

**Reply by minthaka** [reply]
So basically there is no option to achieve from API the same as it is possible using GUI:

![kép](https://user-images.githubusercontent.com/8353588/196250686-81bd039e-266f-4f42-849f-3066b5aafc08.png)

Am I right?

**Reply by harshavardhana** [reply]
It is possible via API its called SetBucketPolicy()
